code/integration-tests/src/androidTest/java/com/adobe/marketing/mobile/integration/signal/SignalExtensionIntegrationTests.kt

```diff
@@ -0,0 +1,223 @@
+                countDownLatch.countDown()
+            }
+            countDownLatch.await(100, TimeUnit.MILLISECONDS)
+
+        }
+    }
+
+    @Before
+    fun setUP() {
+
+    }
+
+    @Test
+    fun testGetRequest() {
+        val countDownLatch = CountDownLatch(1)
+        networkMonitor = { url ->
+            if (url == "https://www.signal.com?name=testGetRequest") {
+                countDownLatch.countDown()
+            }
+        }
+        MobileCore.updateConfiguration(
+            mapOf(
+                "global.privacy" to "optedin",
+                "rules.url" to "https://adobe.com/rules_signal.zip"
+            )
+        )
+        Thread.sleep(10)
+        MobileCore.dispatchEvent(
+            Event.Builder("Test", "type", "source")
+                .setEventData(
+                    mapOf(
+                        "name" to "testGetRequest"
+                    )
+                ).build()
+        )
+        assertTrue(countDownLatch.await(1, TimeUnit.SECONDS))
+    }
+
+    @Test
+    fun testPostRequest() {
+        val countDownLatch = CountDownLatch(1)
+        networkMonitor = { url ->
+            if (url == "https://www.signal.com?name=testPostRequest") {
+                countDownLatch.countDown()
+            }
+        }
+        MobileCore.updateConfiguration(
+            mapOf(
+                "global.privacy" to "optedin",
+                "rules.url" to "https://adobe.com/rules_signal.zip"
+            )
+        )
+        Thread.sleep(10)
+        MobileCore.dispatchEvent(
+            Event.Builder("Test", "type", "source")
+                .setEventData(
+                    mapOf(
+                        "name" to "testPostRequest"
+                    )
+                ).build()
+        )
+        assertTrue(countDownLatch.await(1, TimeUnit.SECONDS))
+    }
+
+    @Test
+    fun testOptedOut() {
+        var signalRequestCaught = false
+        val countDownLatch = CountDownLatch(1)
+        networkMonitor = { url ->
+            if (url.startsWith("https://www.signal.com")) {
+                signalRequestCaught = true
+            }
+        }
+        MobileCore.updateConfiguration(
+            mapOf(
+                "global.privacy" to "optedout",
+                "rules.url" to "https://adobe.com/rules_signal.zip"
+            )
+        )
+        Thread.sleep(10)
+        MobileCore.dispatchEvent(
+            Event.Builder("Test", "type", "source")
+                .setEventData(
+                    mapOf(
+                        "name" to "testPostRequest"
+                    )
+                ).build()
+        )
+        MobileCore.dispatchEvent(
+            Event.Builder("Test", "type", "source")
+                .setEventData(
+                    mapOf(
+                        "name" to "testGetRequest"
+                    )
+                ).build()
+        )
+        countDownLatch.await(10, TimeUnit.MILLISECONDS)
```

Comment:
nit: could we assertFalse here to ensure the thread wasn't interrupted before wait time elapsed?